src/main/java/org/axonframework/springboot/nativex/AxonRuntimeHints.java

```diff
@@ -0,0 +1,238 @@
+import org.axonframework.eventhandling.ReplayToken;
+import org.axonframework.eventhandling.tokenstore.ConfigToken;
+import org.axonframework.eventsourcing.EventSourcingHandler;
+import org.axonframework.messaging.responsetypes.InstanceResponseType;
+import org.axonframework.messaging.responsetypes.MultipleInstancesResponseType;
+import org.axonframework.messaging.responsetypes.OptionalResponseType;
+import org.axonframework.modelling.saga.SagaEventHandler;
+import org.axonframework.modelling.saga.repository.jpa.SagaEntry;
+import org.axonframework.queryhandling.QueryHandler;
+import org.axonframework.spring.stereotype.Aggregate;
+import org.axonframework.spring.stereotype.Saga;
+import org.springframework.aot.hint.MemberCategory;
+import org.springframework.aot.hint.ReflectionHints;
+import org.springframework.aot.hint.RuntimeHints;
+import org.springframework.aot.hint.RuntimeHintsRegistrar;
+import org.springframework.aot.hint.TypeHint;
+import org.springframework.aot.hint.TypeReference;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider;
+import org.springframework.core.type.filter.AnnotationTypeFilter;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.reflect.Method;
+import java.sql.Connection;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * This class is used to set runtime hints for axon applications. It finds the root package, and starts from there to
+ * search vor axon classes, and add reflection hints. From those classes it tries to find all the api classes, which
+ * need to have there constructors and methods available for serialisation. It does a few additional things which are
+ * needed to make it work.
+ *
+ * @author Gerard Klijs
+ * @since 4.8.0
+ */
+public class AxonRuntimeHints implements RuntimeHintsRegistrar {
+
+    private static final String PACKAGE_SEPARATOR = ".";
+    private static final String FOLDER_SEPARATOR = "/";
+
+    @Override
+    public void registerHints(RuntimeHints hints, ClassLoader classLoader) {
+        ReflectionHints reflectionHints = hints.reflection();
+        registerGrpcHints(reflectionHints);
+        List<Class<?>> axonClasses = getAxonClasses(classLoader);
+        registerReflectionHints(axonClasses, reflectionHints);
+        Set<Class<?>> apiClasses = new HashSet<>(axonSerializableClasses());
+        addClassesUsedInAggregateConstructors(apiClasses, axonClasses);
+        addClassesUsedInHandlers(apiClasses, axonClasses);
+        apiClasses.forEach(c -> registerForSerialisation(reflectionHints, c));
```

Comment:
Nice clean solution here, I like.
Also curious to see we are required to add classes used in constructors and handlers separately. Good find!